internal/bootfiles/bootfiles.go

```diff
@@ -55,3 +60,30 @@ func (b *BootFiles) DownloadBootFiles(ctx context.Context, params operations.Dow
 	return filemiddleware.NewResponder(operations.NewDownloadBootFilesOK().WithPayload(reader),
 		objectName, contentLength)
 }
+
+func (b *BootFiles) UploadBootFiles(ctx context.Context, params operations.UploadBootFilesParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	// Create/Update version in DB
+	version := *params.UploadBootFilesParams.OpenshiftVersion
```

Comment:
keep in mind that if the version already exist the controller will need all the details in order for him to create a cluster (openshift version) so you can just call for `Save` you will need to get the object and return all the values.